src/surface.cpp

```diff
@@ -1303,10 +1330,18 @@ void read_surfaces(pugi::xml_node node)
     // condition.  Otherwise, it is a rotational periodic BC.
     if (std::abs(1.0 - dot_prod) < FP_PRECISION) {
       surf1.bc_ = std::make_shared<TranslationalPeriodicBC>(i_surf, j_surf);
-      surf2.bc_ = surf1.bc_;
+      surf2.bc_ = std::make_shared<TranslationalPeriodicBC>(i_surf, j_surf);
```

Comment:
If periodic surfaces don't share a BoundaryCondition object, there is no need or them to be `shared_ptr` (could be `unique_ptr` instead). You don't necessarily need to make that change as part of this PR but just wanted to point this out while it's still in my mind so we can come back to it later.

Reply:
I'll fix this shortly